homeassistant/components/screenlogic/entity.py

```diff
@@ -0,0 +1,142 @@
+        """Return device information for the controller."""
+        controller_type = self.config_data["controller_type"]
+        hardware_type = self.config_data["hardware_type"]
+        try:
+            equipment_model = EQUIPMENT.CONTROLLER_HARDWARE[controller_type][
+                hardware_type
+            ]
+        except KeyError:
+            equipment_model = f"Unknown Model C:{controller_type} H:{hardware_type}"
+        return DeviceInfo(
+            connections={(dr.CONNECTION_NETWORK_MAC, self.mac)},
+            manufacturer="Pentair",
+            model=equipment_model,
+            name=self.gateway_name,
+            sw_version=self.gateway.version,
+        )
+
+    async def _async_refresh(self):
+        """Refresh the data from the gateway."""
+        await self.coordinator.async_refresh()
+        # Second debounced refresh to catch any secondary
+        # changes in the device
+        await self.coordinator.async_request_refresh()
+
+    async def _async_refresh_timed(self, now):
+        """Refresh from a timed called."""
+        await self.coordinator.async_request_refresh()
+
+
+class ScreenLogicPushEntity(ScreenlogicEntity):
+    """Base class for all ScreenLogic push entities."""
+
+    def __init__(self, coordinator, data_key, message_code, enabled=True):
+        """Initialize the entity."""
+        super().__init__(coordinator, data_key, enabled)
+        self._update_message_code = message_code
+
+    @callback
+    def _async_data_updated(self) -> None:
+        """Handle data updates."""
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+
+        self.async_on_remove(
+            await self.gateway.async_subscribe_client(
+                self._async_data_updated, self._update_message_code
+            )
+        )
+
+
+class ScreenLogicCircuitEntity(ScreenLogicPushEntity):
+    """Base class for all ScreenLogic switch and light entities."""
+
+    _attr_has_entity_name = True
+
+    @property
+    def name(self):
+        """Get the name of the switch."""
+        return self.circuit["name"]
+
+    @property
+    def is_on(self) -> bool:
+        """Get whether the switch is in on state."""
+        return self.circuit["value"] == ON_OFF.ON
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Send the ON command."""
+        await self._async_set_circuit(ON_OFF.ON)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Send the OFF command."""
+        await self._async_set_circuit(ON_OFF.OFF)
+
+    async def _async_set_circuit(self, circuit_value) -> None:
+        if await self.gateway.async_set_circuit(self._data_key, circuit_value):
+            _LOGGER.debug("Turn %s %s", self._data_key, circuit_value)
+        else:
+            raise HomeAssistantError(
+                f"Failed to set_circuit {self._data_key} {circuit_value}"
+            )
```

Comment:
```suggestion
         _LOGGER.debug("Turn %s %s", self._data_key, circuit_value)
        if not await self.gateway.async_set_circuit(self._data_key, circuit_value):
            raise HomeAssistantError(
                f"Failed to set_circuit {self._data_key} {circuit_value}"
            )
```

Does it make sense to always log the call?

Reply:
It can potentially help with debugging calls and tracking steps from HA through to the api. The climate entity is also set up this way.
`screenlogicpy` intentionally has A LOT of debug logging at the moment, especially as there's a bunch of new code and I'm still discovering oddities with SL's comms depending on pool config.
Client push implementation I think is the last major re-work of the api. I imagine this type of logging could be removed ~v1.0 of the api.
Can remove if still desired.

Reply:
For clarity I was suggesting moving the logger outside of the if so it always logs.  It wasn't clear to me that it was logging a success state or a failure state.  Or maybe move it after the `if not` in my suggested change and make it clear it was successful?

Reply:
Ah, sorry. I misunderstood. Yea, it was intentionally logging the success. I can move it out after the `if not`.

Reply:
I'll make the same change in `climate` as well.